metadata-db-app/src/cmr/metadata_db/data/oracle/generic_documents.clj

```diff
@@ -124,31 +124,30 @@
 (defn save-concept
   "Create the document in the database, try to pull as many values out of the
    document as can be found. All documents must have at least a :Name and a
-   :MetadataSpecification field."
+   :MetadataSpecification field inside the metadata."
   [db provider-id document]
-  (let [rm-fields [:deleted :concept-id :concept-type :created-at :native-id :provider-id :revision-date :revision-id]
-        metadata (json/generate-string (strip-fields document rm-fields))
-        metadata-spec (get document :MetadataSpecification)
-        schema (get metadata-spec :Name)
+  (let [metadata (json/parse-string (:metadata document) true)
+        metadata-spec (get metadata :MetadataSpecification)
+        schema (string/lower-case (get metadata-spec :Name))
         version (get metadata-spec :Version)
         created-at (coerce/to-sql-time (dtp/parse-datetime (:created-at document)))
-        revision-date (coerce/to-sql-time (dtp/parse-datetime (:revision-date document)))]
-    (insert-record db
-                   {:id (get-next-id-seq db)
-                    :concept_id (:concept-id document)
-                    :native_id (:native-id document)
-                    :provider_id provider-id
-                    :document_name (get document :Name)
-                    :schema schema
-                    :format schema
-                    :mime_type (format "application/%s;version=%s" schema version)
-                    :metadata (cutil/string->gzip-bytes metadata)
-                    :revision_id (:revision-id document)
-                    :revision_date revision-date
-                    :created_at created-at
-                    :deleted (or (:deleted document) 0)
-                    :user_id "place-holder"
-                    :transaction_id (get-next-transaction-id db)})))
+        revision-date (coerce/to-sql-time (dtp/parse-datetime (:revision-date document)))
+        db-record {:id (get-next-id-seq db)
+                   :concept_id (:concept-id document)
+                   :native_id (:native-id document)
+                   :provider_id provider-id
+                   :document_name (:document-name document)   ;(subs (:document-name metadata) 0 20)
```

Comment:
Do we need the commented out line at the end?  ;(subs ...

Reply:
it should be handled by the schema, I will remove